canal/canal_test.go

```diff
@@ -323,3 +323,93 @@ func TestWithoutSchemeExp(t *testing.T) {
 		}
 	}
 }
+
+func TestCreateIndexExp(t *testing.T) {
+	cases := []string{
+		"create index test0 on test.test (id)",
+		"create index test0 ON test.test (id)",
+		"CREATE INDEX test0 on `test`.test (id)",
+		"CREATE INDEX test0 ON test.test (id)",
+		"CREATE index test0 on `test`.test (id)",
+		"CREATE index test0 ON test.test (id)",
+		"create INDEX test0 on `test`.test (id)",
+		"create INDEX test0 ON test.test (id)",
+		"CREATE INDEX `test0` ON `test`.`test` (`id`) /* generated by server */",
+		"CREATE /*generated by server */ INDEX `test0` ON `test`.`test` (`id`)",
+		"CREATE INDEX `test0` ON `test`.test (id)",
+		"CREATE INDEX `test0` ON test.`test` (id)",
+		"CREATE INDEX `test0` ON test.test (`id`)",
+		"CREATE INDEX test0 ON `test`.`test` (`id`)",
+		"CREATE INDEX test0 ON `test`.`test` (id)",
+		"CREATE INDEX test0 ON test.test (`id`)",
+	}
+
+	baseTable := "test"
+	db := "test"
+	pr := parser.New()
+	for _, s := range cases {
+		stmts, _, err := pr.Parse(s, "", "")
+		require.NoError(t, err)
+		for _, st := range stmts {
+			nodes := parseStmt(st)
+			if len(nodes) == 0 {
+				t.Fatalf("TestCreateIndexExp: parse fail\n")
+				continue
+			}
+			for _, node := range nodes {
+				rdb := node.db
+				rtable := node.table
+				if (len(rdb) != 0 && rdb != db) || rtable != baseTable {
```

Comment:
and here, you can use `require.Equal(t, db, rdb)` and `require.Equal(t, baseTable, rtable)`

Reply:
I have apply require package to test